src/Symfony/Component/Workflow/Event/HasContextTrait.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Event;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Hugo Hamon <hugohamon@neuf.fr>
+ *
+ * @internal
+ */
+trait HasContextTrait
+{
+    private array $context = [];
```

Comment:
I made the property `private` instead of `protected` as all concrete classes using this trait are already all `final`. Although it may be a potential BC break, it should not as event classes are final and internal. Application code should are not supposed to extend the base abstract `Event` class to produce new subtypes of workflow events.